### PR TITLE
Reapply "Reapply ABI generation outside of `/dist` (#1157)" (#1158)

### PR DESCRIPTION
**Pull request by iamacook**
This reverts commit 73f5941a340d7e9c8032f26f0a1e30ff50ca32b5, generating ABIs from `/abis` and bundling them in Docker.

The `postinstall` script was dependent on package changes and, as such, is not included before the `build` script.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,13 +4,15 @@
...
+    "build": "yarn generate-abis && nest build",
     "format": "prettier --write .",
     "format-check": "prettier --check .",
+    "generate-abis": "node ./scripts/generate-abis.js",
+    "postinstall": "yarn generate-abis",
     "start": "nest start",
     "start:dev": "nest start --watch",
     "start:debug": "nest start --debug --watch",
-    "start:prod": "node dist/main",
+    "start:prod": "node dist/src/main",
```
Comment (iamacook) [inline package.json:15]: By importing from a folder outside of `src`, Nest creates a "nested" `dist`, e.g.:

```
└── dist
    ├── abis
    └── src
```

File: Dockerfile [hunk 2]
```diff
@@ -28,8 +29,9 @@ ARG BUILD_NUMBER
 ENV APPLICATION_VERSION=${VERSION} \
     APPLICATION_BUILD_NUMBER=${BUILD_NUMBER}
 
+COPY --chown=node:node --from=base /app/abis ./abis
 COPY --chown=node:node --from=base /app/node_modules ./node_modules
 COPY --chown=node:node --from=base /app/dist ./dist
 COPY --chown=node:node --from=base /app/assets ./assets
 COPY --chown=node:node --from=base /app/migrations ./migrations
-CMD [ "node", "dist/main.js" ]
+CMD [ "node", "dist/src/main.js" ]
```
Comment (iamacook) [inline Dockerfile:37]: By importing from a folder outside of `src`, Nest creates a "nested" `dist`, e.g.:

```
└── dist
    ├── abis
    └── src
```

File: package.json (hunk 1 above)
Comment (iamacook) [inline package.json:15]: As above regarding the "nested" `dist`.

File: package.json [hunk 3]
```diff
@@ -4,13 +4,15 @@
   "private": true,
   "license": "MIT",
   "scripts": {
-    "build": "nest build",
+    "build": "yarn generate-abis && nest build",
     "format": "prettier --write .",
     "format-check": "prettier --check .",
+    "generate-abis": "node ./scripts/generate-abis.js",
+    "postinstall": "yarn generate-abis",
```
Comment (hectorgomezv) [inline package.json:11]: Is this still needed (`generate-abis` is already called on the build step)?

Comment (iamacook) [reply package.json:11]: I think so - if you clone the project, you need the ABIs to run it. I considered adding it to the `start` scripts too but thought it was too much.

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**